lib/uri/common.rb

```diff
@@ -314,17 +314,40 @@ def self.regexp(schemes = nil)
   TBLDECWWWCOMP_['+'] = ' '
   TBLDECWWWCOMP_.freeze
 
-  # Encodes given +str+ to URL-encoded form data.
+  # Returns a URL-encoded string derived from the given string +str+.
   #
-  # This method doesn't convert *, -, ., 0-9, A-Z, _, a-z, but does convert SP
-  # (ASCII space) to + and converts others to %XX.
+  # The returned string:
   #
-  # If +enc+ is given, convert +str+ to the encoding before percent encoding.
+  # - Preserves:
   #
-  # This is an implementation of
-  # https://www.w3.org/TR/2013/CR-html5-20130806/forms.html#url-encoded-form-data.
+  #   - Characters <tt>'*'</tt>, <tt>'.'</tt>, <tt>'-'</tt>, and <tt>'_'</tt>.
+  #   - Character in ranges <tt>'a'..'z'</tt>, <tt>'A'..'Z'</tt>,
+  #     and <tt>'0'..'9'</tt>.
+  #
+  #   Example:
+  #
+  #     URI.encode_www_form_component('*.-_azAZ09')
+  #     # => "*.-_azAZ09"
+  #
+  # - Converts:
+  #
+  #   - Character <tt>' '</tt> to character <tt>'+'</tt>.
+  #   - Any other character to "percent notation";
+  #     the percent notation for character <i>c</i> is <tt>'%%%x' % c.ord</tt>.
```

Comment:
It seems to encode to hex in upper case.

```suggestion
  #     the percent notation for character <i>c</i> is <tt>'%%%X' % c.ord</tt>.
```

Reply:
Fixed.